### ISSUE TITLE
Weird looking nerd icons on St terminal 

### ISSUE THREAD

**Issue by mortezadadgar** [jarun/nnn#969]
#### Environment details

[x] Operating System: Archlinux
[x] Desktop Environment: bspwm
[x] Terminal Emulator: st
[x] Shell: zsh
[ ] Custom desktop opener (if applicable):
[ ] Program options used:
[ ] Configuration options set:
[ ] Issue exists on `nnn` master
![image](https://user-images.githubusercontent.com/27911493/116084085-ec003180-a68c-11eb-814e-ec61037334fc.png)
Also st error about invalid characters on opening nnn:
`erresc: unknown csi ESC[23;0t
 erresc: unknown csi ESC[22;0t`

**Comment by KlzXS** [comment]
I don't know what those escape sequences mean, couldn't find them on wikipedia, but I myself have never seen such an error on st.

I know of the icon issue, but I don't know of a reliable way to fix it. The best thing I found is to add extra spaces after it as having pther characters "too close" to it can make it look like that. The place to add padding is [here](https://github.com/jarun/nnn/blob/eea4d34423f3b1aabebcd36ba707a25a4780f990/src/icons.h#L25). There should already be a signle space, but try two.

**Comment by jarun** [comment]
@mortezadadgar are you using the option `-x`?

@KlzXS these are from printing the xterm title to terminal. And I have no plans to add and option to toggle xterm title printing.

**Comment by jarun** [comment]
@luukvbaal can you please take a look?

**Comment by mortezadadgar** [comment]
> Regd. the prints...
> 
> @mortezadadgar are you using the option `-x`?
> 
> @KlzXS these are from printing the xterm title to terminal. And I have no plans to add and option to toggle xterm title printing.

It's not related to -x, same issue after removing the option

**Comment by jarun** [comment]
And you are not using sessions, right?

**Comment by jarun** [comment]
I am pretty sure these are from xterm title printing as there prints are:

https://github.com/jarun/nnn/blob/eea4d34423f3b1aabebcd36ba707a25a4780f990/src/nnn.c#L5938

https://github.com/jarun/nnn/blob/eea4d34423f3b1aabebcd36ba707a25a4780f990/src/nnn.c#L7540

**Comment by mortezadadgar** [comment]
> And you are not using sessions, right?
No

**Comment by mortezadadgar** [comment]
I think it's related to how St manges icons overlaps
St:
![image](https://user-images.githubusercontent.com/27911493/116091787-b4958300-a694-11eb-9442-6cdaaa44ad7d.png)
Alacritty:
![image](https://user-images.githubusercontent.com/27911493/116091837-c2e39f00-a694-11eb-9021-2817b6113ce5.png)

**Comment by KlzXS** [comment]
> I think it's related to how St manges icons overlaps

That's right. We can't do anything about that. `nnn` just sends text for the terminal to render, it's up to the terminal to decide how it should do that. Try adding extra padding to avoid overlap, that's the best advice I can give you.

**Comment by luukvbaal** [comment]
You can fix this issue with the patch mentioned [here].

**Comment by jarun** [comment]
Cool! Closing the defect.
